mempool/src/shared_mempool.rs

```diff
@@ -600,6 +606,18 @@ where
     }
 }
 
+fn process_config_update(config_update: ConfigUpdate) {
+    // check for VM update
+    if let Some(_bytes) = config_update.get(&VMPublishingOption::config_id()) {
```

Comment:
can't we instead of operating on hashmap level(ConfigUpdate) create OnChainConfig struct with strongly typed methods and pass that around?

components won't have to do deserialization themself. It will be something like `config.get_vm_publishing_option()` in their callbacks

also you won't have to do unions of subscriptions. Just collect whatever is inside of such object